pronto/api/entry/references.py

```diff
@@ -186,3 +186,149 @@ def unlink_reference(accession, pub_id):
+            }
+        }), 401
+
+    sig_query = """
+        SELECT METHOD_AC
+        FROM INTERPRO.ENTRY2METHOD
+        WHERE ENTRY_AC = :1
+    """
+
+    con = utils.connect_oracle_auth(user)
+    with con.cursor() as cur:
+        cur.execute(sig_query, [accession])
+        signature_accs = cur.fetchall()
+
+    # process each signature in turn & batch process proteins for each signature
+    signature_accs = [_[0] for _ in signature_accs]
+    invalid_pmids, failed_pub_ids = update_signature_citations(
+        signature_accs, accession, con
+    )
+
+    con.close()
+
+    if len(invalid_pmids) == len(failed_pub_ids) == 0:
+        return jsonify({
+            "status": True,
+        }), 200
+    else:
+        return jsonify({
+            "status": False,
+            "error": {
+                "title": "Failed citation update",
+                "message": f"Failed to update citations and/or relate citations to entry {accession}.",
+                "invalid_pmids": invalid_pmids,
+                "failed_pub_ids": failed_pub_ids,
+            }
+        }), 400
+
+
+def update_signature_citations(
+    sig_accs: list[str],
+    entry_acc: str,
+    orc_con
+) -> tuple[list[str], list[str]]:
+    """Coordinate inserting missing citations from PostGreSQL (e.g. INTTST) db 
+    into the IPRO db (e.g. IPDEV), and relate the new citations to the InterPro entry
+
+    :param sig_accs: list of signature accs associated with interpro entry
+    :param entry_acc: str, IPPRO entry accession
+    :param orc_con: open connection to IPPRO oracle db
+    """
+    invalid_pmids, failed_pub_ids = [], []
+    pmid_query = f"""
+        SELECT P.pubmed_id
+        FROM protein2publication P
+        INNER JOIN signature2protein S ON P.protein_acc = S.protein_acc
+        WHERE S.signature_acc IN ({','.join(['%s'] * len(sig_accs))})
+    """
+
+    pg_con = utils.connect_pg(utils.get_pg_url())
+
+    with pg_con.cursor() as pg_cur:
+        pg_cur.execute(pmid_query, sig_accs)
+        while True:
+            pmid_batch = pg_cur.fetchmany(size=1000)
+            if not pmid_batch:
+                break
+
+            not_in_oracle = check_pmid_in_citations([_[0] for _ in pmid_batch], orc_con)
+
+            if len(not_in_oracle) > 0:
+                with orc_con.cursor() as cit_cur:
+                    new_citations = get_citations(cit_cur, not_in_oracle)
+                    error = insert_citations(cit_cur, new_citations)
+
+                if error:
+                    invalid_pmids.extend(list(new_citations.keys()))
+
+                else:
+                    for pmid, pub_id in new_citations.items():
+                        error = relate_entry_to_pubs(entry_acc, pub_id, orc_con)
+                        if error:
+                            failed_pub_ids.append(pmid)
+
+    pg_con.close()
+    orc_con.commit()
+
+    return invalid_pmids, failed_pub_ids
+
+
+def check_pmid_in_citations(pmids: list[str], orc_con) -> list[int]:
```

Comment:
Fix type in type hinting.

```suggestion
def check_pmid_in_citations(pmids: list[int], orc_con) -> list[int]:
```